src/main/java/duke/commands/FindCommand.java

```diff
@@ -0,0 +1,40 @@
+package duke.commands;
+
+import duke.tasks.TaskList;
+import duke.util.Storage;
+import duke.util.Ui;
+
+import java.util.List;
+
+public class FindCommand extends Command {
+    String keyword;
+
+    public FindCommand(String keyword) {
+        this.keyword = keyword;
+    }
+
+    public boolean isExit() {
+        return false;
+    }
+
+    public void execute(TaskList taskList, Ui ui, Storage storage) {
+        TaskList results = taskList.search(keyword);
+        String toPrint = "";
+        if (results.numTasks() > 0) {
+            toPrint += "Here are the matching tasks in your list!!\n";
+            for (int i = 0; i < results.numTasks(); i++) {
+                toPrint += "\t(" + (i + 1) + ") " + results.getList().get(i) + "\n";
+            }
+        } else if (results.numTasks() == 0) {
+            toPrint += "There are no matching results in your list.";
+        } else if (taskList.numTasks() == 0) {
+            toPrint += "You have nothing on your list.";
+        }
+        System.out.println(Ui.setBorder(toPrint));
+    }
+
+    @Override
+    public boolean equals(Object command) {
+        return command instanceof FindCommand;
+    }
+}
```

Comment:
I was wondering why did you choose to store the string to be printed in the variable toPrint and printing it at the end of the method instead of directly printing it. While there's not much difference, it doesn't seem to be necessary to use a variable here.

Reply:
I can't remember why I did this, but thanks for the comments! I will probably change it later, thank you :)